app/controllers/videos_controller.rb

```diff
@@ -0,0 +1,15 @@
+class VideosController < ApplicationController
+  before_action :require_user
+  def index
+    @videos = Video.all
```

Comment:
Are you using `@videos` in your `app/views/videos/index.html.haml` ? If not then this is redundant, You can easily get to all videos from `@categories` when you iterate of them.